src/main/scala/org/broadinstitute/dsde/agora/server/webservice/methods/MethodsServiceHandlers.scala

```diff
@@ -1,25 +1,29 @@
 package org.broadinstitute.dsde.agora.server.webservice.methods
 
 import akka.actor.Actor
-import org.broadinstitute.dsde.agora.server.webservice.MethodsQueryResponse
+import org.broadinstitute.dsde.agora.server.dataaccess.AgoraDao
+import org.broadinstitute.dsde.agora.server.model.AgoraEntity
 import org.broadinstitute.dsde.agora.server.webservice.util.ServiceMessages
 import spray.routing.RequestContext
+import com.novus.salat._
+import com.novus.salat.global._
+
 
 
 /**
- * Actor responsible for querying unmapped bam information from the Vault
+ * Actor responsible for querying the methods repository.
  */
 class MethodsQueryHandler extends Actor {
   implicit val system = context.system
 
   def receive = {
-    case ServiceMessages.Query(requestContext: RequestContext, id: String) =>
-      query(requestContext, id)
+    case ServiceMessages.Query(requestContext: RequestContext, namespace: String, name: String, id: Int) =>
+      query(requestContext, namespace, name, id)
       context.stop(self)
   }
-  def query(requestContext: RequestContext, id: String): Unit = {
 
-    // TODO- Business logic to retrieve a method from the storage implementation
-    context.parent ! MethodsQueryResponse(id, Map("cwlFile" -> "fake/file/url"), Map("author" -> "Dave Shiga", "version" -> "1.0"))
+  def query(requestContext: RequestContext, namespace: String, name: String, id: Int): Unit = {
+    val method = AgoraDao.createAgoraDao.findSingle(namespace, name, id)
+    context.parent ! grater[AgoraEntity].toJSON(method)  
```

Comment:
We still have to figure out marshalling in the perRequest, yah?

Could we just have complete() in that class expect from everyone, keeping it general, and delegate serialization to the service handler actors as we do here?